custom_components/tuya_local/devices/confortotal_minisplit.yaml

```diff
@@ -0,0 +1,175 @@
+name: Air conditioner
+products:
+  - id: mzhwsoeaumuxleaf
+    name: Confortotal Mini Split CICON242
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: auto
+              value: heat_cool
+            - dps_val: cold
+              value: cool
+            - dps_val: wind
+              value: fan_only
+            - dps_val: wet
+              value: dry
+            - dps_val: hot
+              value: heat
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 16
+        max: 32
+      unit: C
+      mapping:
+        - step: 1
+    - id: 3
+      type: integer
+      name: current_temperature
+      unit: F
```

Comment:
The set temp is sending data in F but returning current temp in C when the machine turns on and this was the only way for me to get around it. If I leave the unit as blank or C and set the temp to 61F, it will return the data as 61C which converts to 142F.

Reply:
Here is a post with similar problem.

https://community.home-assistant.io/t/finally-got-a-pioneer-heat-pump-working-in-ha-via-local-tuya-with-some-hacks/499372

Reply:
Upon further testing after removing the unit from current_temperature and leaving temperature unit as C, the unit continues to report everything in C which HA will convert to F on the card, this is expected behavior which is okay. What is weird is that my remote set to display target temp in F and when I use HA to control the unit, the machine will display the target temp in C (i.e 16C if I set the temp to 61F) but if turn on the machine with the remote, the machine will display in F (61F) and reporting the target temp as C (61C) making HA convert the temp to 142F. What is the recommend way to get around this?

The current_temperature is always reporting in C so there are no issue there.

I have made removed the unit from current_temperature for now.

Reply:
This sounds like a similar issue as beca_bhp6000_thermostat_mapped.yaml
Maybe the same hack will work here?

Reply:
Upon testing via Tuya's developer platform. The temp set accepts value from 16 to 90. If the temp is set between 16 - 32, it will set the display to C. If it's 61 - 90, it will be set the display to F. If it's between 33 and 60, it will be set to 61.

```
code: temp_set
type: Integer
value:
{
  "unit": "℃",
  "min": 16,
  "max": 90,
  "scale": 0,
  "step": 1
}
```

Updated DPS 2 with the following and got close. The temperture in HA matches what is on the machine in both C and F.

``` 
    - id: 2
      type: integer
      name: temperature
      range:
        min: 16
        max: 32
      mapping:
        - dps_val: 61
          value: 16.1
        - dps_val: 62
          value: 16.7
        - dps_val: 63
          value: 17.2
        - dps_val: 64
          value: 17.8
        - dps_val: 65
          value: 18.3
        - dps_val: 66
          value: 18.9
        - dps_val: 67
          value: 19.4
        - dps_val: 68
          value: 20
        - dps_val: 69
          value: 20.6
        - dps_val: 70
          value: 21.1
        - dps_val: 71
          value: 21.7
        - dps_val: 72
          value: 22.2
        - dps_val: 73
          value: 22.8
        - dps_val: 74
          value: 23.3
        - dps_val: 75
          value: 23.9
        - dps_val: 76
          value: 24.4
        - dps_val: 77
          value: 25
        - dps_val: 78
          value: 25.6
        - dps_val: 79
          value: 26.1
        - dps_val: 80
          value: 26.7
        - dps_val: 81
          value: 27.2
        - dps_val: 82
          value: 27.8
        - dps_val: 83
          value: 28.3
        - dps_val: 84
          value: 28.9
        - dps_val: 85
          value: 29.4
        - dps_val: 86
          value: 30
        - dps_val: 87
          value: 30.6
        - dps_val: 88
          value: 31.1
        - dps_val: 89
          value: 31.7
        - dps_val: 90
          value: 32.2
```

However, I am not unable to set the temp in HA unless I change the range from 16 - 32 to 16 - 90 which will cause HA's thermostat to have a range of 61F to 194F. What would be the suggested fix?

2024-04-21 10:10:02.085 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Air conditioner: Mapped dps 2 to 62 from 16.666666666666668
2024-04-21 10:10:02.085 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Considering mapping for range of temperature
2024-04-21 10:10:02.085 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Considering mapping for range of temperature
2024-04-21 10:10:02.085 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140614959430720] temperature (16.666666666666668) must be between 16.0 and 32.0
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 239, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2543, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2580, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 971, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1043, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 917, in async_service_temperature_set
    await entity.async_set_temperature(**kwargs)
  File "/config/custom_components/tuya_local/climate.py", line 240, in async_set_temperature
    await self.async_set_target_temperature(
  File "/config/custom_components/tuya_local/climate.py", line 252, in async_set_target_temperature
    await self._temperature_dps.async_set_value(
  File "/config/custom_components/tuya_local/helpers/device_config.py", line 469, in async_set_value
    settings = self.get_values_to_set(device, value)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tuya_local/helpers/device_config.py", line 939, in get_values_to_set
    raise ValueError(f"{self.name} ({value}) must be between {mn} and {mx}")
ValueError: temperature (16.666666666666668) must be between 16.0 and 32.0

Reply:
Added the following to as a hack to lock max temp to 90F for now until there is a better solution.

/config/configuration.yaml
```
homeassistant:
  customize: !include customize.yaml
```

/config/customize.yaml

```
climate.entity_id:
    max_temp: 90
```